library/common.h

```diff
@@ -149,4 +150,34 @@ inline void mbedtls_xor(unsigned char *r, const unsigned char *a, const unsigned
 #endif
 /* *INDENT-ON* */
 
+/* Always provide a static assert macro, so it can be used unconditionally.
+ * Note that it will expand to nothing on some systems.
+ * Can be used outside functions (but don't add a trailing ';' in that case:
+ * the semicolon is included here to avoid triggering -Wextra-semi when
+ * MBEDTLS_STATIC_ASSERT() expands to nothing).
+ * Can't use the C11-style `defined(static_assert)` on FreeBSD, since it
```

Comment:
Oh, you didn't find a way to bypass than warning :disappointed: . Maybe we could detect FreeBSD and use `_Static_assert` there?

Reply:
`_Static_assert()` is only available depending what you're building for. But yes, that could be added in a follow-up PR